plugins/registry/handler.go

```diff
@@ -200,6 +209,28 @@ func (rh *registryHandler) relay(ctx context.Context, p registryHandlerParams) {
 		}
 	}
 
+	// reject the push request if the model namespace does not already exist in model-backend
+	ctx = withUserUIDAuth(ctx, p.userUID)
+	if isOrganizationRepository {
```

Comment:
❓ Does it make sense to perform the check only on a given operation? There will be several blob / manifest requests, so checking the existence of the namespace in each one of them might be redundant and add some overhead.

Reply:
This is the sequence of calls, there's no "initial call" so we can reduce the client calls by performing them on the HEAD or POST operation, but there will be > 1. We might also consider introducing a cache 

```
[KRAKEND] 2024/04/01 - 11:11:42.347 ▶ DEBUG ---> HEAD /v2/admin/hello-world/blobs/sha256:478afc9190022e867bb857b1a25cc5abc7678287af6cb930562ec25be709f1b7
[KRAKEND] 2024/04/01 - 11:11:42.356 ▶ DEBUG ---> POST /v2/admin/hello-world/blobs/uploads/
[KRAKEND] 2024/04/01 - 11:11:42.363 ▶ DEBUG ---> PATCH /v2/admin/hello-world/blobs/uploads/9016e453-0208-4fad-b2b3-910d618820db
[KRAKEND] 2024/04/01 - 11:11:42.370 ▶ DEBUG ---> PUT /v2/admin/hello-world/blobs/uploads/9016e453-0208-4fad-b2b3-910d618820db
[KRAKEND] 2024/04/01 - 11:11:42.376 ▶ DEBUG ---> HEAD /v2/admin/hello-world/blobs/sha256:478afc9190022e867bb857b1a25cc5abc7678287af6cb930562ec25be709f1b7
[KRAKEND] 2024/04/01 - 11:11:42.381 ▶ DEBUG ---> HEAD /v2/admin/hello-world/blobs/sha256:ee301c921b8aadc002973b2e0c3da17d701dcd994b606769a7e6eaa100b81d44
[KRAKEND] 2024/04/01 - 11:11:42.384 ▶ DEBUG ---> POST /v2/admin/hello-world/blobs/uploads/
[KRAKEND] 2024/04/01 - 11:11:42.388 ▶ DEBUG ---> PATCH /v2/admin/hello-world/blobs/uploads/b245d491-e78b-48ed-abb2-0101c1081030
[KRAKEND] 2024/04/01 - 11:11:42.393 ▶ DEBUG ---> PUT /v2/admin/hello-world/blobs/uploads/b245d491-e78b-48ed-abb2-0101c1081030
[KRAKEND] 2024/04/01 - 11:11:42.397 ▶ DEBUG ---> HEAD /v2/admin/hello-world/blobs/sha256:ee301c921b8aadc002973b2e0c3da17d701dcd994b606769a7e6eaa100b81d44
[KRAKEND] 2024/04/01 - 11:11:42.400 ▶ DEBUG ---> PUT /v2/admin/hello-world/manifests/0.1.0
```

Reply:
indeed, we should definitely avoid the redundant calls to model-backend, will refactor